api/ipfsproxy/ipfsproxy.go

```diff
@@ -135,6 +139,16 @@ func New(cfg *Config) (*Server, error) {
 		IdleTimeout:       cfg.IdleTimeout,
 		Handler:           router,
 	}
+	if cfg.Tracing {
```

Comment:
s.Handler right above should be set to this . You'll need to move this right after `router := mux.NewRouter()` i think

Reply:
Sorry, I misread. I think it would be clearer if `router` modified before creating the server, rather than modifying the handler after creating the server.